std/assembly/util/number.ts

```diff
@@ -559,16 +559,26 @@ function getCachedPower(minExp: i32): void {
 
 // @ts-ignore: decorator
 @inline
-function grisu2(value: f64, buffer: usize, sign: i32): i32 {
+function grisu2(value: f64, buffer: usize, sign: i32, isSinglePrecision: bool): i32 {
+  let frc: u64;
+  let exp: i32;
 
   // frexp routine
-  let uv  = reinterpret<u64>(value);
-  let exp = i32((uv & 0x7FF0000000000000) >>> 52);
-  let sid = uv & 0x000FFFFFFFFFFFFF;
-  let frc = (u64(exp != 0) << 52) + sid;
-  exp = select<i32>(exp, 1, exp) - (0x3FF + 52);
+  if (isSinglePrecision) {
+    let uv = reinterpret<u32>(<f32>value);
+    exp = (uv & 0x7F800000) >>> 23;
+    let sid = uv & 0x007FFFFF;
+    frc = (u64(exp != 0) << 23) + sid;
+    exp = select<i32>(exp, 1, exp) - (0x7F + 23);
```

Comment:
We can simplify `select<i32>(exp, 1, exp)` to `exp || 1` nowadays.
```suggestion
    exp = (exp || 1) - (0x7F + 23);
```